packages/shared/components/atoms/tiles/TransactionActivityTile.svelte

```diff
@@ -30,51 +42,70 @@
+    $: $selectedAccountAssets, (asset = getAssetFromPersistedAssets(data.assetId))
+
+    $: isUnclaimed = data.asyncStatus === ActivityAsyncStatus.Unclaimed
+    $: isTimelocked = data.asyncStatus === ActivityAsyncStatus.Timelocked
+    $: isIncoming = data.direction === ActivityDirection.In
+    $: isIncomingActivityUnclaimed = (isIncoming || data.isSelfTransaction) && isUnclaimed
+    $: isAsyncActivity = isTimelocked || (data.isAsync && (data.direction === ActivityDirection.Out || isUnclaimed))
+    $: {
+        if (isUnclaimed || isTimelocked) {
+            const activityDirectionKey = isIncoming ? 'incoming' : 'outgoing'
+            const activityAsyncStatusKey = isUnclaimed ? 'expirationTime' : 'timelockDate'
+            const textKey = `tooltips.transactionDetails.${activityDirectionKey}.${activityAsyncStatusKey}`
+            asyncStatusTooltipText = localize(textKey)
+        }
+    }
 
     function getTimeDiff(txData: ITransactionActivityData): string {
-        if (isTimelocked) {
-            return getTimeDifference(txData.timelockDate, $time)
-        } else if (txData.isAsync && !txData.isClaimed && txData?.expirationDate) {
-            return getTimeDifference(txData.expirationDate, $time)
-        } else {
-            return localize('general.none')
+        const { asyncStatus, isAsync, isClaimed, expirationDate, timelockDate } = txData
+
+        if (asyncStatus === ActivityAsyncStatus.Timelocked) {
+            return getTimeDifference(timelockDate, $time)
+        }
+        if (isAsync && !isClaimed && expirationDate) {
+            return getTimeDifference(expirationDate, $time)
         }
+        return localize('general.none')
     }
 
     function getTitle(txData: ITransactionActivityData, inclusionState: InclusionState): string {
-        if (txData.isShimmerClaiming) {
-            return inclusionState === InclusionState.Confirmed ? 'general.shimmerClaimed' : 'general.shimmerClaiming'
-        } else if (txData.isInternal) {
-            return inclusionState === InclusionState.Confirmed ? 'general.transfer' : 'general.transferring'
-        } else {
-            if (txData.direction === ActivityDirection.In) {
-                return inclusionState === InclusionState.Confirmed ? 'general.received' : 'general.receiving'
-            } else if (txData.direction === ActivityDirection.Out) {
-                return inclusionState === InclusionState.Confirmed ? 'general.sent' : 'general.sending'
-            }
+        const { isShimmerClaiming, isInternal, direction } = txData
+        const isInclusionStateConfirmed = inclusionState === InclusionState.Confirmed
+
+        if (isShimmerClaiming) {
+            return isInclusionStateConfirmed ? 'general.shimmerClaimed' : 'general.shimmerClaiming'
+        }
+        if (isInternal) {
+            return isInclusionStateConfirmed ? 'general.transfer' : 'general.transferring'
+        }
+        if (direction === ActivityDirection.In) {
```

Comment:
All values inside this function rely on the function scope instead of the global scope, I think this change could mix contexts and make it a little bit harder to debug/understand, even though I like your suggestion as it is easier to read.